rotkehlchen/chain/ethereum/modules/yearn/decoder.py

```diff
@@ -93,97 +107,131 @@ def reload_data(self) -> None:
             )
             cursor.execute(f'SELECT COUNT(*) {query_body}', bindings)
             if cursor.fetchone()[0] == len(self.vaults[YEARN_VAULTS_V3_PROTOCOL]) + len(self.vaults[YEARN_VAULTS_V2_PROTOCOL]) + len(self.vaults[YEARN_VAULTS_V1_PROTOCOL]):  # noqa: E501
-                return  # up to date
+                return None  # up to date
 
             # we are missing new vaults. Populate the cache
             cursor.execute(f'SELECT protocol, address {query_body}', bindings)
             for row in cursor:
                 self.vaults[row[0]].add(string_to_evm_address(row[1]))
 
+        return self.addresses_to_decoders()
+
     def _maybe_enrich_yearn_transfers(
             self,
             context: EnricherContext,
     ) -> TransferEnrichmentOutput:
         """
-        Enrich ethereum transfers made during the execution of yearn contracts.
+        Enrich ethereum transfers made during the execution of yearn v1 and some v2 contracts.
         This enricher detects:
         - deposits
         - withdrawals
 
-        First we make sure that the contract is a yearn v1, v2 or v3 contract and that the method
+        First we make sure that the contract is a yearn v1 or v2 contract and that the method
         executed in the contract is one of the expected.
         """
-        if context.transaction.to_address is None:
+        if (
+            context.tx_log.topics[0] != TRANSFER_TOPIC or
+            context.transaction.to_address is None
+        ):
             return FAILED_ENRICHMENT_OUTPUT
 
-        protocol = CPT_YEARN_V3
+        protocol = YEARN_VAULTS_V2_PROTOCOL
         if context.transaction.to_address in self.vaults[YEARN_VAULTS_V1_PROTOCOL]:
-            protocol = CPT_YEARN_V1
-        elif context.transaction.to_address in self.vaults[YEARN_VAULTS_V2_PROTOCOL]:
-            protocol = CPT_YEARN_V2
-        elif context.transaction.to_address not in self.vaults[YEARN_VAULTS_V3_PROTOCOL]:
+            protocol = YEARN_VAULTS_V1_PROTOCOL
+        elif context.transaction.to_address not in self.vaults[YEARN_VAULTS_V2_PROTOCOL]:
             return FAILED_ENRICHMENT_OUTPUT
 
-        is_deposit = False
         if (
             context.transaction.input_data.startswith(YEARN_DEPOSIT_4_BYTES) or
-            context.transaction.input_data.startswith(YEARN_DEPOSIT_AMOUNT_4_BYTES) or
-            context.transaction.input_data.startswith(YEARN_DEPOSIT_4_BYTES_V3)
+            context.transaction.input_data.startswith(YEARN_DEPOSIT_AMOUNT_4_BYTES)
```

Comment:
why are you checking this here? Input data decoding is not reliable. And even if you want to use this then you should use `decoding_by_input_data`, not here.

Reply:
Unfortunately it seems that the input data is the only way to identify these transfer events. I've reworked it to use `decoding_by_input_data` instead of enriching.